wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationLegalHoldCell.swift

```diff
@@ -77,32 +77,45 @@ final class ConversationLegalHoldCellDescription: ConversationMessageCellDescrip
     }
 
     private static func configuration(for systemMessageType: ZMSystemMessageType, in conversation: ZMConversation) -> View.Configuration {
-
-        let baseTemplate = "content.system.message_legal_hold"
-        var template = baseTemplate
+        let systemMessageTitle = title(for: systemMessageType)
+        var attributedText = NSAttributedString(string: systemMessageTitle, attributes: ConversationSystemMessageCell.baseAttributes)
 
         if systemMessageType == .legalHoldEnabled {
-            template += ".enabled"
-        } else if systemMessageType == .legalHoldDisabled {
-            template += ".disabled"
+            attributedText = appendLegalHoldEnabledText(to: attributedText)
         }
 
-        var attributedText = NSAttributedString(string: template.localized, attributes: ConversationSystemMessageCell.baseAttributes)
+        let icon = StyleKitIcon.legalholdactive.makeImage(size: .tiny, color: SemanticColors.Icon.foregroundDefaultRed)
 
-        if systemMessageType == .legalHoldEnabled {
-            let learnMore = NSAttributedString(string: (baseTemplate + ".learn_more").localized.uppercased(),
-                                               attributes: [.font: UIFont.mediumSemiboldFont,
-                                                            .link: ConversationLegalHoldSystemMessageCell.legalHoldURL as AnyObject,
-                                                            .foregroundColor: SemanticColors.Label.textDefault])
+        return View.Configuration(attributedText: attributedText, icon: icon, conversation: conversation)
+    }
 
-            attributedText += " " + String.MessageToolbox.middleDot + " "
-            attributedText += learnMore
+    private static func title(for messageType: ZMSystemMessageType) -> String {
```

Comment:
I think there is no need to make this method as a static and the same for` configuration(for ...) `method.

Reply:
It needs to be static otherwise we will get this error in line 

```swift
init(systemMessageType: ZMSystemMessageType, conversation: ZMConversation) {
        configuration = ConversationLegalHoldCellDescription.configuration(for: systemMessageType, in: conversation) ERROR: // Instance member 'configuration' cannot be used on type 'ConversationLegalHoldCellDescription'; did you mean to use a value of this type instead?
        accessibilityLabel = configuration.attributedText?.string
    }
```

Reply:
After a bit of more digging and understanding I came to the conclusion that  these methods do not rely on or alter the state of an instance of their class. They are designed to work with the parameters passed to them (systemMessageType and conversation) rather than instance variables. This makes them ideal candidates for static methods since they can be called without needing to instantiate the class.

So I guess It's fine to have them as static.